tests/e2e/workloads/bdi/bdi_base_class.py

```diff
@@ -0,0 +1,232 @@
+            chart_dir=self.chart_dir,
+        )
+        create_secretes_factory(
+            db2u_project_name=self.db2u_project,
+            ldap_r_n=self.ldap_r_n,
+            ldap_r_p=self.ldap_r_p,
+            db2u_r_n=self.db2u_r_n,
+            db2u_r_p=self.db2u_r_p,
+        )
+        create_ibm_container_registry_factory(
+            db2u_project_name=self.db2u_project, ibm_cloud_key=self.ibm_cloud_key
+        )
+
+    def run(self):
+        """
+        Installing db2warehouse, running the workload and perform a cleanup for temporary files
+        """
+        now = datetime.now()
+        current_time = now.strftime("%H:%M:%S")
+        f"##################### Start setup at: {current_time} #######################"
+        install_db2u(self)
+        now = datetime.now()
+        current_time = now.strftime("%H:%M:%S")
+        log.info(
+            f"##################### Finished setup at: {current_time} #######################"
+        )
+        self.run_workload()
+        clean_temp_files(self)
+
+    def run_workload(self):
+        """Running the workload:
+        This function instantiates and creates 3 yaml files for creating 3 jobs:
+        - configure workload - Creates database and all required tables as well as generates the data.
+        - data load - loads the data to the DB tables.
+        - run workload - runs 3 iterations for reading/writing from/to the DB tables:
+            - warmup
+            - static values, multiuser (16 users)
+            - static values, multiuser (32 users)
+        """
+
+        now = datetime.now()
+        current_time = now.strftime("%H:%M:%S")
+        f"##################### Start configure-workload at: {current_time} #######################"
+        ocp_jobs = ocp.OCP(kind="Job", namespace=self.db2u_project)
+
+        # CONFIGURE WORKLOAD
+        self.temp_configure_dict = init_configure_workload_yaml(
+            namespace=self.db2u_project,
+            image=self.db2u_image_url,
+            sf=self.scale_factor,
+            pvc_name=self.db2u_pvc_name,
+        )
+
+        self.temp_yaml_configure = tempfile.NamedTemporaryFile(
+            mode="w", prefix="configure_yaml_", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            self.temp_configure_dict, self.temp_yaml_configure.name
+        )
+
+        log.info("Creating Job for for configure-workload")
+        ocp_jobs.create(yaml_file=self.temp_yaml_configure.name)
+        ocp_jobs.wait_for_resource(
+            condition="1/1",
+            resource_name="db2u-configure-workload-job",
+            column="COMPLETIONS",
+            timeout=self.configure_timeout,
+            sleep=30,
+        )
+        now = datetime.now()
+
+        current_time = now.strftime("%H:%M:%S")
+        log.info(
+            f"##################### Finished configure-workload at: {current_time} #######################"
+        )
+
+        now = datetime.now()
```

Comment:
there is nothing between the 2 datetime() commands - except logging the time, so why not just run 2 log commands ?

Reply:
Correct.
Done